pkg/prometheus/server/operator.go

```diff
@@ -995,6 +1041,9 @@ func (c *Operator) UpdateStatus(ctx context.Context, key string) error {
 	p := pobj.(*monitoringv1.Prometheus)
 	p = p.DeepCopy()
 
+	if p == nil || c.rr.DeletionInProgress(p) {
```

Comment:
i suppose that you've hit a race condition? Should we be more defensive in DeletionInProgress (e.g. if p is nil then it would return true)? If yes, the fix could deserve its own PR :)